Dockerfile

```diff
@@ -25,35 +27,33 @@ RUN cp -r build/* /app/static
 RUN mv /app/static/static/* /app/static/
 RUN rm -r /app/static/static
 
+# Set up Django
+FROM python:alpine
 
-FROM python:buster
+# Get PostgreSQL client
+# https://stackoverflow.com/a/47871121
+RUN apk update
+RUN apk add --no-cache postgresql-libs
+RUN apk add --no-cache --virtual build-dependencies gcc musl-dev postgresql-dev
 
-# Copy contents from builder
-COPY --from=builder /app/static /app/static
+# Copy static site from webapp
+COPY --from=webapp /app/static /app/static
 
-# Copy base files
-COPY banwebScrape.py /app/
-COPY course_gather /app/course_gather
-COPY manage.py /app/
-COPY requirements.txt /app/
-COPY scraper /app/scraper
+WORKDIR /app
 
-WORKDIR /app/
+COPY requirements.txt .
 
-# Set environment variables
-ARG DJANGO_SETTINGS_MODULE
-ENV DJANGO_SETTINGS_MODULE=$DJANGO_SETTINGS_MODULE
+# Install Python dependencies
+RUN pip3 install --upgrade pip -r requirements.txt --no-cache-dir
+
+# Remove unneeded build dependencies for smaller image
+RUN apk del build-dependencies
 
-ARG SECRET_KEY
-ENV SECRET_KEY=$SECRET_KEY
+# Set environment variables
+ENV DJANGO_SETTINGS_MODULE="course_gather.settings.prod_settings"
 
-# Install python deps and build backend
-RUN pip3 install --upgrade pip -r requirements.txt
-RUN python3 ./manage.py makemigrations
-RUN python3 ./manage.py migrate
-RUN python3 ./banwebScrape.py
-RUN python3 ./manage.py custom_db_sync /app/writer.csv
+# Copy source
+COPY . .
```

Comment:
I think we should keep this a bit more explicit. To me this raises the question of "what all is coming into my environment?"
For example, I don't think a prod Dockerfile needs the docs/ folder. Nor does it need the webapp folder.

Reply:
I can add `docs/` to `.dockerignore`, but to me it seems easier to copy everything by default and ignore a few files/directories in `.dockerignore` instead of having 5 different COPY commands. With explicit COPY commands, every time a run-time file/directory dependency is added the Dockerfile has to be updated as well. Just would make it possible to accidentally get in a bad state where someone adds a dependency but forgets to update the Dockerfile.

If you don't think the top-level dependencies will change that often I can revert back to the separate COPY commands.

Reply:
I don't really see much top level changes atm. The new models are going to be under course_gather/ and anything UI related is under webapp/ (which we may want to rename at some point?, idk)

Reply:
Alright, reverted for now.